be/test/fs/fs_starlet_test.cpp

```diff
@@ -0,0 +1,248 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include <fmt/format.h>
+#include <fs/fs_starlet.h>
+#include <gtest/gtest.h>
+#include <starlet.h>
+
+#include <fstream>
+
+#include "common/config.h"
+#include "gutil/strings/join.h"
+#include "testutil/assert.h"
+
+namespace starrocks {
+
+extern staros::starlet::Starlet* g_starlet;
+
+class StarletFileSystemTest : public testing::Test {
+public:
+    StarletFileSystemTest() = default;
+    ~StarletFileSystemTest() override = default;
+    void SetUp() override {
+        _starlet = new staros::starlet::Starlet();
+        _starlet->init(8888);
+        g_starlet = _starlet;
+    }
+    void TearDown() override {
+        g_starlet = nullptr;
+        delete _starlet;
+    }
+
+    std::string StarletPath(std::string_view path) {
+        if (path.front() == '/') {
+            return fmt::format("staros://s3://{}.s3.{}.{}{}", kBucket, kRegion, kDomain, path);
```

Comment:
not sure what's the final design is. but this `staros://s3://` scheme is a non-standard URL format, and is hardly to use to be compatible to leverage other common libraries for URL parsing and manipulation. `staros+s3://`, or `staros_s3://` will be preferred if it is a must have design.